src/core/pointcloud/qgslazinfo.cpp

```
@@ -0,0 +1,305 @@
+  mAttributes.push_back( QgsPointCloudAttribute( "EdgeOfFlightLine", QgsPointCloudAttribute::Char ) );
+  mAttributes.push_back( QgsPointCloudAttribute( "Classification", QgsPointCloudAttribute::Char ) );
+  mAttributes.push_back( QgsPointCloudAttribute( "ScanAngleRank", QgsPointCloudAttribute::Short ) );
+  mAttributes.push_back( QgsPointCloudAttribute( "UserData", QgsPointCloudAttribute::Char ) );
+  mAttributes.push_back( QgsPointCloudAttribute( "PointSourceId", QgsPointCloudAttribute::UShort ) );
+
+  if ( mPointFormat == 6 || mPointFormat == 7 || mPointFormat == 8 || mPointFormat == 9 || mPointFormat == 10 )
+  {
+    mAttributes.push_back( QgsPointCloudAttribute( "ScannerChannel", QgsPointCloudAttribute::Char ) );
+    mAttributes.push_back( QgsPointCloudAttribute( "ClassificationFlags", QgsPointCloudAttribute::Char ) );
+    mAttributes.push_back( QgsPointCloudAttribute( "GpsTime", QgsPointCloudAttribute::Double ) );
+  }
+  if ( mPointFormat == 1 || mPointFormat == 3 || mPointFormat == 4 || mPointFormat == 5 )
+  {
+    mAttributes.push_back( QgsPointCloudAttribute( "GpsTime", QgsPointCloudAttribute::Double ) );
+  }
+  if ( mPointFormat == 2 || mPointFormat == 3 || mPointFormat == 5 || mPointFormat == 7 || mPointFormat == 8 || mPointFormat == 10 )
+  {
+    mAttributes.push_back( QgsPointCloudAttribute( QStringLiteral( "Red" ), QgsPointCloudAttribute::UShort ) );
+    mAttributes.push_back( QgsPointCloudAttribute( QStringLiteral( "Green" ), QgsPointCloudAttribute::UShort ) );
+    mAttributes.push_back( QgsPointCloudAttribute( QStringLiteral( "Blue" ), QgsPointCloudAttribute::UShort ) );
+  }
+  if ( mPointFormat == 8 || mPointFormat == 10 )
+  {
+    mAttributes.push_back( QgsPointCloudAttribute( QStringLiteral( "NIR" ), QgsPointCloudAttribute::UShort ) );
+  }
+  if ( mPointFormat == 4 || mPointFormat == 5 || mPointFormat == 9 )
+  {
+    // TODO: handle the Wave Packets attributes
+  }
```

I would suggest to remove any support for point formats 4,5,9,10 because PDAL does not support full waveform data, so we're not going to support them anytime soon either :-)

We should actually have a check in QgsLazInfo when parsing header - if the point format is not 0,1,2,3,6,7,8 then we should abort (better to fail early than having random bugs due to unsupported data format)

---

We can just not support the wave packets handling and keep the formats readable? QGIS will behave like the attributes of wave packets don't exist at all.